File: sentimental-credit/credit.py
# TODO
import re
import cs50
import sys

gg = 0


def main():
    lol = cs50.get_int("Number: ")
    lol = str(lol)
    if len(lol) == 15:
        AMEX(lol)
    elif len(lol) == 13:
        VISA(lol)
    elif len(lol) ==16:
        help(lol)
    else:
        invalid()


def AMEX(lol):
    for i in [34,37]:
        a = str(i)
        pattern = re.compile(a)
        if pattern.search(lol, 0, 2) != None:

            if checksum(lol) == True:
                print("AMEX")
                sys.exit(0)

    invalid()

def invalid():
    print("INVALID")
    sys.exit(0)

def VISA(lol):
    for i in [4]:
        a = str(i)
        pattern = re.compile(a)
        if pattern.search(lol, 0) != None:
                if checksum(lol) == True:
                    print("VISA")
                    sys.exit(0)
    invalid()

def MAC(lol):
    for i in [51,52,53,54,55]:
        a = str(i)
        pattern = re.compile(a)
        if pattern.search(lol, 0, 2) != None:
                if checksum(lol) == True:
                    print("MASTERCARD")
                    sys.exit(0)
    invalid()

def help(lol):
    if lol[0] == "4":
        VISA(lol)
    else:
        MAC(lol)

def checksum(lol):
    check = 0
    bro = len(lol)
    see = int(lol)

    for i in range(bro):
        if (i + 1)%2 == 0:
            z = 2*(see%10)
            if z >= 10:
                z = 1 + z%10
        else:
            z = see%10
        check += z
        see = see//10

    if check%10 == 0:
        return True
    else:
        return False




if __name__ == "__main__":
    main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>


long bro(long z);


int main(void)
{
long x;


        x = get_long("number: ");

    if (x<4000000000000||x>4999999999999&x<340000000000000||x>379999999999999&x<4000000000000000||x>4999999999999999&x<5100000000000000||x>5599999999999999 )
    {
    printf("INVALID\n");
    
}
        //VISA13
        if (x>3999999999999&x<5000000000000)
        {
            int visa13 = bro(x);
            if (visa13%10==0)
            {
                printf("VISA\n");
            }
            else
            {
                printf("INVALID\n");
            }
        }

        //AMEX
          if (x>339999999999999&x<350000000000000)
        {
            int amex = bro(x);
                if (amex%10 == 0)
                {
                    printf("AMEX\n");
                }
                    else
                {
            printf("INVALID\n");
                }
        }
        //AMEX 2
          if (x>369999999999999&x<380000000000000)
        {
            int amex2 = bro(x);
                if (amex2%10 == 0)
                {
                    printf("AMEX\n");
                }
                    else
                {
            printf("INVALID\n");
                }
        }
        //AMEX REJ
        if (x>349999999999999&x<370000000000000)
       {
        printf("INVALID\n");
       }
        // MASTERCARD
            if (x>5099999999999999&x<5600000000000000)
            {
                int mastercard = bro(x);
                    if (mastercard%10==0)
                    {
                        printf("MASTERCARD\n");
                    }
                       else
                    {
            printf("INVALID\n");
                    }
            }
        // VISA16
            if (x>3999999999999999&x<5000000000000000)
            {
                int visa16 = bro(x);
                    if (visa16%10==0)
                    {
                        printf("VISA\n");
                    }

            else
                   {
            printf("INVALID\n");
            }
            }
}

long bro(long z)
{
    long y;
    long a;
    long b;
    long c;
    long d;
    long e;
    long f;
    long g;
    long h;
    long s;




    y = (z/100000000000000%10+z/1000000000000%10+z/10000000000%10+z/100000000%10+z/1000000%10+z/10000%10+z/100%10+z%10);

        if (z/1000000000000000%10*2>=10)
            {
            a = (z/1000000000000000%10*2%10+1);
            }
            else
            {
                 a = (z/1000000000000000%10*2);
            }
        if (z/10000000000000%10*2>=10)
        {
            b = (z/10000000000000%10*2%10+1);
        }
        else
        {
             b = (z/10000000000000%10*2);
        }
        if (z/100000000000%10*2>=10)
        {
             c =(z/100000000000%10*2%10 + 1);
        }
        else
        {
            c=(z/100000000000%10*2);

          }  if (z/1000000000%10*2>=10)
           {
            d=(z/1000000000%10*2%10 + 1);
           }
           else
           {
            d= (z/1000000000%10*2);
           }
        if (z/10000000%10*2>=10)
        {
           e=(z/10000000%10*2%10 + 1);
        }
        else
        {
            e = (z/10000000%10*2);
        }
        if (z/100000%10*2>=10)
        {
            f = (z/100000%10*2%10 + 1);
        }
        else
        {
            f=(z/100000%10*2);
        }
        if (z/1000%10*2>=10)
        {
           g =  (z/1000%10*2%10 + 1 );
        }
        else
        {
            g =  (z/1000%10*2);
        }
        if (z/10%10*2>=10)
        {
            h = (z/10%10*2%10 + 1 );
        }
        else
        {
             h = (z/10%10*2);
        }
           s = y+a+b+c+d+e+f+h+g;

           return s;

}
